lib/http/headers.rb

```diff
@@ -94,7 +94,18 @@ def to_a
     #
     # @return [String]
     def inspect
-      "#<#{self.class} #{to_h.inspect}>"
+      "#<#{self.class} #{pretty_print}>"
+    end
+
+    # Returns a printable representation of headers.
+    #
+    # @param [String] separator
+    # @return [String]
+    def pretty_print(separator = ", ")
+      to_h.map do |name, value|
+        value = value.join("; ") if value.respond_to?(:join)
+        "#{name}: #{value}"
+      end.join(separator)
```

Comment:
I'm not strongly against that. Although I thought it's better to print them just as they would be sent to server (we don't join them), but for pretty printing I think it worth to do so. Also, I think in this case it might worth to allow specify truncation length so that long lines would become truncated (have no opinion on this at all though). Anyway, if you think it worth join headers let' do that, but a bit more efficient way will be:

``` ruby
headers.each.group_by { |x| x[0] }.each do |key, values|
  io << "\n#{key}: #{values.join '; '}"
end
```